src/huggingface_hub/hf_file_system.py

```diff
@@ -43,10 +43,17 @@ class HfFileSystemResolvedPath:
     repo_id: str
     revision: str
     path_in_repo: str
+    # The part placer after '@' in the path (can even be a quoted or unquoted special refs revision)
+    _revision_in_path: Optional[str] = None
```

Comment:
I would rename the attribute if that's fine with you to really emphasize it is the revision as provided by the user. I found `revision_in_path` to be slightly vague. I suggested `_raw_revision` but happy for other suggestions.

(would have to be updated in the PR)

```suggestion
    # The part placed after '@' in the initial path. It can be a quoted or unquoted refs revision.
    # Used to reconstruct the unresolved path to return to the user.
    _raw_revision: Optional[str] = None
```